### PR TITLE
feat(cli): override static files datadir

### LINKED ISSUE #6945: Make it possible to store static files outside of the datadir
### Describe the feature

Currently, we derive the path for static files purely from the node-wide datadir https://github.com/paradigmxyz/reth/blob/b75862010aca90faee8687d969aa10b4076cb947/crates/node-core/src/dirs.rs#L285-L288

Instead, we should be able to pass a separate argument `--datadir.static-files` that overrides the static files path and defaults to the previous behavior inside the `datadir`.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by justcode740**
busy af, so not have time on this for a bit
but beside some doc , should work now, rely on review for nits, 
if u can help solve these great o.w. have to come back wen have time..
resolves #6945 

@shekhirin could you review it? @gakonst this guy is busy af.

### CODE REVIEW

File: crates/node-core/src/dirs.rs [hunk 1]
```diff
@@ -284,7 +290,12 @@ impl<D> ChainPath<D> {
 
     /// Returns the path to the static_files directory for this chain.
     pub fn static_files_path(&self) -> PathBuf {
-        self.0.join("static_files").into()
+        let datadir_args = &self.2;
+        if let Some(static_path) = &datadir_args.static_files_path {
+            static_path.to_path_buf()
```
Comment (shekhirin) [inline crates/node-core/src/dirs.rs:295]:
```suggestion
        if let Some(static_files_path) = &datadir_args.static_files_path {
            static_files_path.to_path_buf()
```

File: crates/node-core/src/args/datadir_arg.rs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+use std::path::PathBuf;
+
+/// Parameters for database configuration
+#[derive(Debug, Args, PartialEq, Eq, Default, Clone)]
+#[command(next_help_heading = "Datadir")]
+pub struct DatadirArgs {
+    /// Set the absolute path to static files
+    #[arg(long = "datadir.static_files", help_heading = "Datadir", value_name = "PATH")]
+    pub static_files_path: Option<PathBuf>,
+}
```
Comment (shekhirin) [inline crates/node-core/src/args/datadir_arg.rs:13]: this should include the datadir argument, so all datadir-related args are in one place, wdyt @mattsse ?

**Review by shekhirin — CHANGES_REQUESTED** [review]

File: crates/node-core/src/args/datadir_arg.rs (hunk 2 above)
Comment (mattsse) [reply crates/node-core/src/args/datadir_arg.rs:13]: that makes sense yes

File: crates/node-core/src/args/mod.rs [hunk 3]
```diff
@@ -51,6 +51,10 @@ pub use dev_args::DevArgs;
 mod pruning_args;
 pub use pruning_args::PruningArgs;
 
+/// DatadirArgs for configuring data storage paths
+mod datadir_arg;
```
[the remark this answers is not in the document]
Comment (onbjerg) [reply crates/node-core/src/args/mod.rs]: 46899c7

File: crates/node-core/src/dirs.rs [hunk 4]
```diff
@@ -10,7 +10,7 @@ use std::{
 };
 
 /// Constructs a string to be used as a path for configuration and db paths.
-pub fn config_path_prefix(chain: Chain) -> String {
+pub fn config_path_prefix(chain: &Chain) -> String {
```
[the remark this answers is not in the document]
Comment (onbjerg) [reply crates/node-core/src/dirs.rs]: 172d19b

File: crates/node-core/src/args/datadir_arg.rs (hunk 2 above)
Comment (onbjerg) [reply crates/node-core/src/args/datadir_arg.rs]: a36025140292e8c959d0dce7cc76fff12ac7ed75

**Comment by shekhirin** [comment]
Works as intended 
```console
➜  reth git:(0x/override_static_datadir) cargo run -- db --chain holesky --datadir ~/Library/Application\ Support/reth/holesky  stats                                                                                 
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.43s
     Running `target/debug/reth db --chain holesky --datadir '/Users/shekhirin/Library/Application Support/reth/holesky' stats`
| Segment | Block Range | Transaction Range | Shape (columns x rows) | Size |
|---------|-------------|-------------------|------------------------|------|
| ------- | ----------- | ----------------- | ---------------------- | ---- |
| Total   |             |                   |                        | 0 B  |


| Table Name                 | # Entries | Branch Pages | Leaf Pages | Overflow Pages | Total Size |
|----------------------------|-----------|--------------|------------|----------------|------------|
| AccountChangeSets          | 69674674  | 149552       | 813634     | 0              | 3.7 GiB    |
| AccountsHistory            | 23464603  | 8912         | 627740     | 4575           | 2.4 GiB    |
| AccountsTrie               | 3071552   | 1769         | 236758     | 0              | 931.7 MiB  |
| BlockBodyIndices           | 1509089   | 41           | 8831       | 0              | 34.7 MiB   |
| BlockOmmers                | 0         | 0            | 0          | 0              | 0 B        |
| BlockWithdrawals           | 1499143   | 958          | 214106     | 0              | 840.1 MiB  |
| Bytecodes                  | 80842     | 69           | 4786       | 166114         | 667.8 MiB  |
| CanonicalHeaders           | 29        | 1            | 2          | 0              | 12 KiB     |
| HashedAccounts             | 38463002  | 17013        | 1077033    | 0              | 4.2 GiB    |
| HashedStorages             | 73357766  | 24389        | 1416292    | 0              | 5.5 GiB    |
| HeaderNumbers              | 1509088   | 506          | 32969      | 0              | 130.8 MiB  |
| HeaderTerminalDifficulties | 29        | 0            | 1          | 0              | 4 KiB      |
| Headers                    | 29        | 1            | 6          | 0              | 28 KiB     |
| PlainAccountState          | 38463002  | 8094         | 735703     | 0              | 2.8 GiB    |
| PlainStorageState          | 73357766  | 26534        | 1344640    | 0              | 5.2 GiB    |
| PruneCheckpoints           | 4         | 0            | 1          | 0              | 4 KiB      |
| Receipts                   | 1228      | 3            | 61         | 40             | 416 KiB    |
| StageCheckpointProgresses  | 1         | 0            | 1          | 0              | 4 KiB      |
| StageCheckpoints           | 12        | 0            | 1          | 0              | 4 KiB      |
| StorageChangeSets          | 99867933  | 206829       | 1517009    | 0              | 6.6 GiB    |
| StoragesHistory            | 29066521  | 27328        | 1177160    | 2188           | 4.6 GiB    |
| StoragesTrie               | 6079323   | 52025        | 420964     | 0              | 1.8 GiB    |
| TransactionBlocks          | 1333963   | 40           | 8552       | 0              | 33.6 MiB   |
| TransactionHashNumbers     | 51244887  | 10886        | 862829     | 0              | 3.3 GiB    |
| TransactionSenders         | 33473229  | 1399         | 312835     | 0              | 1.2 GiB    |
| Transactions               | 1228      | 4            | 91         | 294            | 1.5 MiB    |
| VersionHistory             | 4         | 0            | 1          | 0              | 4 KiB      |
| -------------------------- | --------- | ------------ | ---------- | -------------- | ---------- |
| Tables                     |           |              |            |                | 44 GiB     |
| Freelist                   | 426014    |              |            |                | 1.6 GiB    |

➜  reth git:(0x/override_static_datadir) cargo run -- db --chain holesky --datadir ~/Library/Application\ Support/reth/holesky --datadir.static_files ~/Library/Application\ Support/reth/holesky/custom_static_files_dir s
tats
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.42s
     Running `target/debug/reth db --chain holesky --datadir '/Users/shekhirin/Library/Application Support/reth/holesky' --datadir.static_files '/Users/shekhirin/Library/Application Support/reth/holesky/custom_static_fi
les_dir' stats`
| Segment      | Block Range | Transaction Range | Shape (columns x rows) | Size      |
|--------------|-------------|-------------------|------------------------|-----------|
| Headers      | 0..=1509088 | N/A               | 3 x 1509089            | 705.3 MiB |
| Transactions | 0..=1509088 | 0..=51244886      | 1 x 51244887           | 37 GiB    |
| Receipts     | 0..=1509088 | 0..=51244886      | 1 x 51244887           | 5.8 GiB   |
| ------------ | ----------- | ----------------- | ---------------------- | --------- |
| Total        |             |                   |                        | 43.5 GiB  |


| Table Name                 | # Entries | Branch Pages | Leaf Pages | Overflow Pages | Total Size |
|----------------------------|-----------|--------------|------------|----------------|------------|
| AccountChangeSets          | 69674674  | 149552       | 813634     | 0              | 3.7 GiB    |
| AccountsHistory            | 23464603  | 8912         | 627740     | 4575           | 2.4 GiB    |
| AccountsTrie               | 3071552   | 1769         | 236758     | 0              | 931.7 MiB  |
| BlockBodyIndices           | 1509089   | 41           | 8831       | 0              | 34.7 MiB   |
| BlockOmmers                | 0         | 0            | 0          | 0              | 0 B        |
| BlockWithdrawals           | 1499143   | 958          | 214106     | 0              | 840.1 MiB  |
| Bytecodes                  | 80842     | 69           | 4786       | 166114         | 667.8 MiB  |
| CanonicalHeaders           | 29        | 1            | 2          | 0              | 12 KiB     |
| HashedAccounts             | 38463002  | 17013        | 1077033    | 0              | 4.2 GiB    |
| HashedStorages             | 73357766  | 24389        | 1416292    | 0              | 5.5 GiB    |
| HeaderNumbers              | 1509088   | 506          | 32969      | 0              | 130.8 MiB  |
| HeaderTerminalDifficulties | 29        | 0            | 1          | 0              | 4 KiB      |
| Headers                    | 29        | 1            | 6          | 0              | 28 KiB     |
| PlainAccountState          | 38463002  | 8094         | 735703     | 0              | 2.8 GiB    |
| PlainStorageState          | 73357766  | 26534        | 1344640    | 0              | 5.2 GiB    |
| PruneCheckpoints           | 4         | 0            | 1          | 0              | 4 KiB      |
| Receipts                   | 1228      | 3            | 61         | 40             | 416 KiB    |
| StageCheckpointProgresses  | 1         | 0            | 1          | 0              | 4 KiB      |
| StageCheckpoints           | 12        | 0            | 1          | 0              | 4 KiB      |
| StorageChangeSets          | 99867933  | 206829       | 1517009    | 0              | 6.6 GiB    |
| StoragesHistory            | 29066521  | 27328        | 1177160    | 2188           | 4.6 GiB    |
| StoragesTrie               | 6079323   | 52025        | 420964     | 0              | 1.8 GiB    |
| TransactionBlocks          | 1333963   | 40           | 8552       | 0              | 33.6 MiB   |
| TransactionHashNumbers     | 51244887  | 10886        | 862829     | 0              | 3.3 GiB    |
| TransactionSenders         | 33473229  | 1399         | 312835     | 0              | 1.2 GiB    |
| Transactions               | 1228      | 4            | 91         | 294            | 1.5 MiB    |
| VersionHistory             | 4         | 0            | 1          | 0              | 4 KiB      |
| -------------------------- | --------- | ------------ | ---------- | -------------- | ---------- |
| Tables                     |           |              |            |                | 44 GiB     |
| Freelist                   | 426014    |              |            |                | 1.6 GiB    |
```

**Review by mattsse — COMMENTED** [review]
this might clash with @joshieDo cli changes

should we proceed with this first @joshieDo ?

**Review by shekhirin — APPROVED** [review]

**Review by joshieDo — APPROVED** [review]

File: crates/node-core/src/dirs.rs [hunk 5]
```diff
@@ -261,12 +267,12 @@ impl<D> From<PathBuf> for MaybePlatformPath<D> {
 /// Otherwise, the path will be dependent on the chain ID:
 ///  * `<DIR>/<CHAIN_ID>`
 #[derive(Clone, Debug, PartialEq, Eq)]
-pub struct ChainPath<D>(PlatformPath<D>, Chain);
+pub struct ChainPath<D>(PlatformPath<D>, Chain, DatadirArgs);
```
Comment (rkrasiuk) [inline crates/node-core/src/dirs.rs:270]: worth making fields named at this point

**Review by rkrasiuk — APPROVED** [review]

**Pull request merged**